forum/jwt_token_middleware.py

```
@@ -0,0 +1,66 @@
+import traceback
+from urllib.parse import parse_qs
+
+from channels.auth import AuthMiddlewareStack
+from channels.db import database_sync_to_async
+from django.conf import settings
+from django.contrib.auth.models import AnonymousUser
+from django.db import close_old_connections
+from jwt import decode as jwt_decode
+from jwt import InvalidSignatureError, ExpiredSignatureError, DecodeError
+
+from authentication.models import CustomUser
+
+
+class JWTAuthMiddleware:
+    """ Checking JWT token for the user"""
+    def __init__(self, app):
+        self.app = app
+
+    async def __call__(self, scope, receive, send):
+        close_old_connections()
+        try:
+            headers = dict(scope['headers'])
+            jwt_token = headers.get(b'authorization', b'').decode("utf-8").split("Bearer ")[-1]
+            if jwt_token:
+                jwt_payload = self.get_payload(jwt_token)
+                user_credentials = self.get_user_credentials(jwt_payload)
+                user = await self.get_logged_in_user(user_credentials)
+                scope['user'] = user
+            else:
+                scope['user'] = AnonymousUser()
+        except (InvalidSignatureError, KeyError, ExpiredSignatureError, DecodeError):
+            traceback.print_exc()
+        except:
```

Catching a broad exception is generally not recommended because it can hide unexpected errors and make debugging difficult. It's better to catch specific exceptions you anticipate

---

fixed